torch/distributed/utils.py

```diff
@@ -155,6 +155,11 @@ def _alloc_storage(tensor: torch.Tensor, size: torch.Size) -> None:
         bool: ``True`` if this method allocated storage and ``False`` if the
         storage was already allocated.
     """
+    # We have made a decision to bypass the free and alloc in FSDP during tracing
+    # partially because having this in the graph won't change anything from a compiling perspective,
+    # and partially because resizing storage in this way is not really a sound thing to do.
+    if torch.distributed._functional_collectives.is_torchdynamo_compiling():
```

Comment:
I think you need to control whether or not you skip by an argument. This file is torch.distributed.utils (not FSDP specific) and it is indeed used in more places than just FSDP.